src/builtins/path.cpp

```diff
@@ -0,0 +1,895 @@
+                         const wgetopter_t &w, options_t *opts) {
+    return handle_flag_perms(argv, parser, streams, w, opts, PERM_WRITE);
+}
+static int handle_flag_x(const wchar_t **argv, parser_t &parser, io_streams_t &streams,
+                         const wgetopter_t &w, options_t *opts) {
+    return handle_flag_perms(argv, parser, streams, w, opts, PERM_EXEC);
+}
+
+static int handle_flag_types(const wchar_t **argv, parser_t &parser, io_streams_t &streams,
+                             const wgetopter_t &w, options_t *opts, path_type_flags_t type) {
+    if (opts->type_valid) {
+        if (!opts->have_type) opts->type = 0;
+        opts->have_type = true;
+        opts->type |= type;
+        return STATUS_CMD_OK;
+    }
+    path_unknown_option(parser, streams, argv[0], argv[w.woptind - 1]);
+    return STATUS_INVALID_ARGS;
+}
+
+static int handle_flag_f(const wchar_t **argv, parser_t &parser, io_streams_t &streams,
+                         const wgetopter_t &w, options_t *opts) {
+    return handle_flag_types(argv, parser, streams, w, opts, TYPE_FILE);
+}
+static int handle_flag_l(const wchar_t **argv, parser_t &parser, io_streams_t &streams,
+                         const wgetopter_t &w, options_t *opts) {
+    return handle_flag_types(argv, parser, streams, w, opts, TYPE_LINK);
+}
+static int handle_flag_d(const wchar_t **argv, parser_t &parser, io_streams_t &streams,
+                         const wgetopter_t &w, options_t *opts) {
+    return handle_flag_types(argv, parser, streams, w, opts, TYPE_DIR);
+}
+
+static int handle_flag_v(const wchar_t **argv, parser_t &parser, io_streams_t &streams,
+                         const wgetopter_t &w, options_t *opts) {
+    if (opts->invert_valid) {
+        opts->invert = true;
+        return STATUS_CMD_OK;
+    }
+    path_unknown_option(parser, streams, argv[0], argv[w.woptind - 1]);
+    return STATUS_INVALID_ARGS;
+}
+
+static int handle_flag_u(const wchar_t **argv, parser_t &parser, io_streams_t &streams,
+                         const wgetopter_t &w, options_t *opts) {
+    if (opts->unique_valid) {
+        opts->unique = true;
+        return STATUS_CMD_OK;
+    }
+    path_unknown_option(parser, streams, argv[0], argv[w.woptind - 1]);
+    return STATUS_INVALID_ARGS;
+}
+
+static int handle_flag_key(const wchar_t **argv, parser_t &parser, io_streams_t &streams,
+                         const wgetopter_t &w, options_t *opts) {
+    UNUSED(argv);
+    UNUSED(parser);
+    UNUSED(streams);
+    opts->have_key = true;
+    opts->key = w.woptarg;
+    return STATUS_CMD_OK;
+}
+
+/// This constructs the wgetopt() short options string based on which arguments are valid for the
+/// subcommand. We have to do this because many short flags have multiple meanings and may or may
+/// not require an argument depending on the meaning.
+static wcstring construct_short_opts(options_t *opts) {  //!OCLINT(high npath complexity)
```

Comment:
I think we can drop this OCLINT comment

Reply:
I'm not sure if oclint is still a thing. I couldn't get it to run last time I tried.

If it's not, we should remove all of these.